### DISCUSSION TITLE
CSS Conflicts Using React + Tailwind App Inside an Existing Ext.JS App

### DISCUSSION THREAD

**Discussion by lucasdebeterco** [tailwindlabs/tailwindcss#12984] [Help]
### **Problem**

I'm loading a React application that uses Tailwind CSS inside an interface (administrative panel) developed with ExtJS, the rendering of the react app was done using a bundle in IIFE format. When the React application is rendered, a <style> tag is added to the top of the page <head>, containing some global Tailwind styling along with the styling needed for the React interface.

Real example, part of global styling created by Tailwind: 

```
body {
  margin:0;
  line-height:inherit;
}

hr {
  height:0;
  color:inherit;
  border-top-width: 1px;
}

h1, h2, h3, h4, h5, h6 {
  font-size:inherit;
  font-weight:inherit;
}

a {
  color:inherit;
  text-decoration:inherit;
}

b,strong {
  font-weight:bolder;
}
```

The globally styled rules of this <style> added to the page, are impacting in several places within this old interface (hr, buttons, font sizes, titles...). Tailwind's global rules are also causing problems with the styling of other React applications that also render in this interface.

Some ways of solving the problem:

### **Apply the CSS rules generated by tailwind only within the box in which the react screen is rendered**

Unfortunately, as far as I've researched, Tailwind doesn't offer a native option to automatically limit your styling to a specific container. A configuration like this would solve part of the problem, making styling restricted to just one area

Example: 

```
#box-central-design body {
  …
}
#box-central-design a {
  …
}

…
```

This solution of wrapping all the rules in a div (#box-central-design), for example, if it worked, would cause another problem. The styling would not apply to modals or any other component that is positioned absolutely on the <body> of page. This component would not be within the #box-central-design, but directly within the body. This action would require a refactoring of the modal components to open a pre-specified element.

### **Remove CSS created by Tailwind when React application window is not visible.**

This is another option that I thought about, but didn't go too deep into it, the main problem in this case is how to add the style back to the page after removing it? Because in the case of changing tabs in the base application (ExtJS), it would be necessary to remove the Tailwind style tag from the page for a certain period until the user returns to the tab containing the React application.

The final solution i applyed was overwriting css hierarchies in all the places that had conflicrts. This solution worked, but i feel is not the best way to fix the problem.

**Finally, is there any way to add a custom class for all CSS rules in the base tailwind styling, as exemplified in the code above? If it's not possible, does anyone have any other ideas or suggestions to attack this problem?**

Thanks!

**Comment by wongjn** [comment]
To add a class scope to all the CSS rule selectors in the base Tailwind styling (named *preflight*), you could consider:
- [Disabling core's preflight plugin](https://tailwindcss.com/docs/preflight#disabling-preflight) and then recreating it with the scope added.
- Having a PostCSS plugin that runs after Tailwind to scope the preflight styles, if you are running Tailwind within a PostCSS pipeline.
